app/controllers/order_items_controller.rb

```diff
@@ -10,7 +11,7 @@ def final_order
       cart = Cart.where(user_id: current_user.id)
       @cart_items = Cartdataitem.where(cart_id: cart)
     end
-    @useraddresses = Useraddress.all
+    @useraddresses = Useraddress.where(user_id: current_user)
```

Comment:
Why you are passing current_user in user_id?
Current_user is object here.

Wrong variable name

Reply:
sure sir.  I will correct this.